src/main/java/dk/xam/starfish/Starfish.java

```diff
@@ -187,22 +213,34 @@ public static void runCommand(Path directory, String... command) throws IOExcept
             .command(command)
             .directory(directory.toFile());
     Process p = pb.start();
+
+    int exit = p.waitFor();
+
+    if (exit != 0) {
+        throw new AssertionError(String.format("runCommand returned %d", exit));
+    }
+    return p;
+}
+
+
+public static String gobbleStream(Process p) throws IOException, InterruptedException{
     StreamGobbler errorGobbler = new StreamGobbler(p.getErrorStream(), "E");
     StreamGobbler outputGobbler = new StreamGobbler(p.getInputStream(), "O");
     outputGobbler.start();
     errorGobbler.start();
     int exit = p.waitFor();
-    errorGobbler.join();
-    outputGobbler.join();
     if (exit != 0) {
         throw new AssertionError(String.format("runCommand returned %d", exit));
     }
+    errorGobbler.join();
+    outputGobbler.join();
+    return outputGobbler.getExecResult()+errorGobbler.getExecResult();
 }
 
 
 
 private static class StreamGobbler extends Thread {
-
+    private volatile String exec_result;
```

Comment:
why volatile ?

Reply:
Just read that "it's used to make classes thread-safe.It means that multiple threads can use a method and instance of the classes at the same time without any problem. example: Assume that two threads are working on the same class. Both threads run on different processors where each thread has its local copy of var. If any thread modifies its value, the change will not reflect in the original one in the main memory. It leads to data inconsistency because the other thread is not aware of the modified value."

But if that's not needed here I could remove that.

Reply:
this class isn't being shared between threads; and new instance created every time. not needed.

Reply:
resolved